egs/ami/s5/cmd.sh

```diff
@@ -12,6 +12,21 @@
 
 export train_cmd="queue.pl --mem 1G"
 export decode_cmd="queue.pl --mem 2G"
+export tfrnnlm_cmd="queue.pl -l hostname=b*"  # this is specific to the CLSP grid
+# we limit to certain machines because TF binaries compiled from C++ take advantage
+# of machine architectures for optimization
+
+export tfrnnlm_py_cmd="queue.pl --gpu 1 CUDA_VISIBLE_DEVICES=\`/home/gkumar/scripts/free-gpu\`"
```

Comment:
Does this actually work?  It's far from clear to me at what stage the backtick command would be evaluated.  I wrote the script queue.pl, but even I don't understand how that works.   It also might not be portable to other similar wrapping scripts (alternatives to queue.pl).  A more generic solution would be to have the TF scripts called from shell scripts which would do this directly (and we should check into the Kaldi repo a version of this free-gpu script, with Gaurav's permission and with appropriate style and copyright headers, etc).  
Actually what might be most convenient from a scripting perspective is to write a script like steps/limit_num_gpus.py which when invoked as;
```
 steps/limit_num_gpus.py  foo --bar=baz
```
would set CUDA_VISIBLE_DEVICES to use just one GPU and would call
```
 foo --bar=baz
```
and when called as
```
 steps/limit_num_gpus.py  --num-gpus=2 foo --bar=baz
```
would do the same but allow 2 GPUs.  (The current script free-gpu does allow you to specify the number of GPUs).
If you rewrite substantially you won't need Gaurav's copyright permission as it will be a rewrite.